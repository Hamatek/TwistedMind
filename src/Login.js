import React from 'react'
import { ScrollView, StyleSheet, TextInput, View } from 'react-native'
// import Icon from 'react-native-vector-icons/FontAwesome'

import Container from '../src/components/Container'
import Button from '../src/components/Button'
import Label from '../src/components/Label'

export default class Login extends React.Component {
  press () {
    // @todo Handles login actions
  }

  render () {
    return (
      <ScrollView style={styles.scroll}>
        <Container>
          {/* <Button
            label='Forgot Login/Pass'
            styles={{
              button: styles.alignRight,
              label: styles.label
            }}
            onPress={this.press.bind(this)}
          /> */}
        </Container>
        <Container>
          <Label text='Username or Email' />
          <TextInput
            autoCapitalize='none'
            autoFocus
            keyboardType='email-address'
            style={styles.textInput}
          />
        </Container>
        <Container>
          <Label text='Password' />
          <TextInput
            secureTextEntry
            style={styles.textInput}
          />
        </Container>
        <View style={styles.footer}>
          <Container>
            <Button
              label='Sign In'
              styles={{
                button: styles.primaryButton,
                label: styles.buttonWhiteText
              }}
              onPress={this.press.bind(this)}
            />
          </Container>
          <Container>
            <Button
              label='CANCEL'
              styles={{label: styles.buttonBlackText}}
              onPress={this.press.bind(this)}
            />
          </Container>
        </View>
      </ScrollView>
    )
  }
}

const styles = StyleSheet.create({
  scroll: {
    backgroundColor: '#E1D7D8',
    padding: 30,
    flexDirection: 'column'
  },
  label: {
    color: '#0d8898',
    fontSize: 20
  },
  alignRight: {
    alignSelf: 'flex-end'
  },
  textInput: {
    height: 80,
    fontSize: 30,
    backgroundColor: '#fff'
  },
  buttonWhiteText: {
    fontSize: 20,
    color: '#fff'
  },
  buttonBlackText: {
    fontSize: 20,
    color: '#595856'
  },
  primaryButton: {
    backgroundColor: '#34A853'
  },
  footer: {
    marginTop: 100
  }
})
