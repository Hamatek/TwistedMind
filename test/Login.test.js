import React from 'react';
import { shallow } from 'enzyme';
import { ScrollView, StyleSheet, Text, TextInput, View } from 'react-native';
import Login from '../src/Login';
// Imports Jest renderer after react-native
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const rendered = renderer.create(<Login />).toJSON();
  expect(rendered).toBeTruthy();
});

it('renders default messages', () => {
  const rendered = renderer.create(<Login />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it('renders two TextInput components', () => {
  const scrollView = shallow(<Login />);
  expect(wrapper.find(TextInput)).to.have.length(2);
});
