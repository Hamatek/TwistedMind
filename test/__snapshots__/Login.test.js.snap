// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default messages 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "#E1D7D8",
      "flexDirection": "column",
      "padding": 30,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    />
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#595856",
            "fontFamily": "sans serif",
            "fontSize": 20,
            "marginBottom": 10,
          }
        }
      >
        Username or Email
      </Text>
      <TextInput
        autoCapitalize="none"
        autoFocus={true}
        keyboardType="email-address"
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 30,
            "height": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#595856",
            "fontFamily": "sans serif",
            "fontSize": 20,
            "marginBottom": 10,
          }
        }
      >
        Password
      </Text>
      <TextInput
        secureTextEntry={true}
        style={
          Object {
            "backgroundColor": "#FFF",
            "fontSize": 30,
            "height": 80,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "marginTop": 100,
        }
      }
    >
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "padding": 20,
                },
                Object {
                  "backgroundColor": "#34A853",
                },
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#FFF",
                "fontSize": 20,
              }
            }
          >
            Sign In
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": 20,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "padding": 20,
                },
                undefined,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#595856",
                "fontSize": 20,
              }
            }
          >
            CANCEL
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
